runtime/v2/shim/publisher.go

```diff
@@ -55,11 +71,42 @@ func (l *remoteEventsPublisher) Close() (err error) {
 	return err
 }
 
-func (l *remoteEventsPublisher) Publish(ctx context.Context, topic string, event events.Event) error {
-	client, err := l.dialer.Get()
-	if err != nil {
-		return err
+func (l *remoteEventsPublisher) processQueue() {
+	for i := range l.requeue {
+		if i.count > maxRequeue {
+			logrus.Errorf("evicting %s from queue because of retry count", i.ev.Topic)
+			// drop the event
+			continue
+		}
+
+		client, err := l.dialer.Get()
+		if err != nil {
+			l.dialer.Put(err)
```

Comment:
Why do we `Put` for `Get` error here, but not at https://github.com/containerd/containerd/pull/3207/files#diff-96e3d1c66f6df0140eab7fed003ce54dR129?